plugins/bigquery/dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -28,14 +29,37 @@
       {% set build_sql = create_table_as(False, target_relation, sql) %}
   {% else %}
      {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
+     
+     {%- set dest_partition = none -%}
+     
+     {#-- if partitioned, get the range of partition values to be updated --#}
+     {% if partition_by %}
+         {%- call statement('pro_tmp', fetch_result = True) -%}
+            {{ create_table_as(True, tmp_relation, sql) }}
+         {%- endcall -%}
+
+         {% call statement('get_partition_range', fetch_result = True) %}
+            select min({{partition_by}}), max({{partition_by}}) from {{tmp_relation}}
+         {% endcall %}
+         
+         {% set partition_range = load_result('get_partition_range').data[0]|list %}
```

Comment:
If the incremental table is empty, then the `min` and `max` columns are going to be null. Can we add a `count(*)` to this query and use that value conditionally supply the `dest_partition` to `get_merge_sql` below? Or, we can just check if the first two columns are null - I don't have a strong preference either way.